src/transformers/convert_graph_to_onnx.py

```diff
@@ -30,14 +16,73 @@ class OnnxConverterArgumentParser(ArgumentParser):
     def __init__(self):
         super(OnnxConverterArgumentParser, self).__init__("ONNX Converter")
 
-        self.add_argument("--pipeline", type=str, choices=SUPPORTED_PIPELINES, default="feature-extraction")
         self.add_argument("--model", type=str, required=True, help="Model's id or path (ex: bert-base-cased)")
         self.add_argument("--tokenizer", type=str, help="Tokenizer's id or path (ex: bert-base-cased)")
         self.add_argument("--framework", type=str, choices=["pt", "tf"], help="Framework for loading the model")
         self.add_argument("--opset", type=int, default=11, help="ONNX opset to use")
         self.add_argument("--check-loading", action="store_true", help="Check ONNX is able to load the model")
         self.add_argument("--use-external-format", action="store_true", help="Allow exporting model >= than 2Gb")
-        self.add_argument("output")
+        self.add_argument("output", type=abspath)
+
+
+if is_torch_available():
+    import torch
+
+    class WrappedModel(torch.nn.Module):
+        def __init__(self, underlying_model):
+            """
+            Class to recursively convert BaseModelOutput* to ONNX supported structs
+            """
+            super(WrappedModel, self).__init__()
+            self.underlying_model = underlying_model
+            self.output_names = {}
+
+        def output_to_tensor(self, output, names=None):
+            if isinstance(output, ModelOutput):
+                parameters = vars(output)
+                output_names = [k for k, v in parameters.items() if v is not None]
+                return self.output_to_tensor(output[:], names=output_names)
+            if isinstance(output, (tuple, list)):
+                output_list = []
+                for o, element in enumerate(output):
+                    element = self.output_to_tensor(element)
+                    # Checks that is a tensor to avoid iterating over it, otherwise flattens the output
+                    if torch.is_tensor(element):
+                        if names is not None:
+                            self.output_names[self.index_output] = names[o]
+                        output_list.append(element)
+                        self.index_output += 1
+                    else:
+                        output_list.extend(element)
+                return output_list
+            return output
+
+        def forward(self, *args, **kwargs):
+            self.index_output = 0
+            output = self.underlying_model(*args, **kwargs)
+            output = self.output_to_tensor(output)
+            return output
+
+
+def get_tokenizer_and_model(tokenizer_name, model_name):
+    tokenizer = AutoTokenizer.from_pretrained(tokenizer_name)
+    model = AutoModel.from_pretrained(model_name)
+    return tokenizer, model
```

Comment:
This is what Pipelines are doing under the hood... I'm still no sure about why we should move away from pipeline? Can you elaborate on this point?

Also, pipeline might see an increased compatibility with ONNX export in the near future. I'm not sure we want to jump in the direction of getting rid of the `Pipeline` representation, it needs to be discussed with the team (cc @LysandreJik, @patrickvonplaten, @sshleifer, @thomwolf, @julien-c)